test/modules/http/response/http-response-readable-stream.test.ts

```diff
@@ -127,19 +132,69 @@ it('forwards ReadableStream errors to the request', async () => {
     response.on('error', responseErrorListener)
   })
 
-  const response = await vi.waitFor(() => {
+  await vi.waitFor(() => {
     return new Promise<http.IncomingMessage>((resolve) => {
-      request.on('response', resolve)
+      request.on('error', resolve)
     })
   })
 
-  // Response stream errors are translated to unhandled exceptions,
-  // and then the server decides how to handle them. This is often
-  // done as returning a 500 response.
-  expect(response.statusCode).toBe(500)
-  expect(response.statusMessage).toBe('Unhandled Exception')
+  // Response body stream errors cannot be translated to unhandled exceptions
+  // since the headers may already have been sent. Instead we need to
+  // ensure that the error event is issued
+
+  expect(responseErrorListener).toHaveBeenCalledOnce()
+  expect(responseErrorListener).toHaveBeenCalledWith(new Error('stream error'))
+
+  expect(requestErrorListener).toHaveBeenCalledOnce()
+  expect(requestErrorListener).toHaveBeenCalledWith(new Error('stream error'))
+  expect(request.destroyed).toBe(true)
+})
+
+it('forwards unhandled ReadableStream errors to the request', async () => {
+  const requestErrorListener = vi.fn()
+  const responseErrorListener = vi.fn()
+  const chunks = ['first', 'second', 'third']
+
+  interceptor.once('request', ({ controller }) => {
+    const stream = new ReadableStream({
+      async pull(controller) {
+        const chunk = chunks.shift()
+        await sleep(200)
+        if (chunk) {
+          controller.enqueue(encoder.encode(chunk))
+        } else {
+          // Enqueue an invalid value
+          controller.enqueue({})
+        }
+      },
+    })
+    controller.respondWith(new Response(stream, {}))
+  })
+
+  const request = http.get('http://localhost/resource')
+  request.on('error', requestErrorListener)
+  request.on('response', (response) => {
+    response.on('error', responseErrorListener)
+  })
+
+  await vi.waitFor(() => {
+    return new Promise<http.IncomingMessage>((resolve) => {
+      request.on('error', resolve)
+    })
+  })
+
+  // Response body stream errors cannot be translated to unhandled exceptions
+  // since the headers may already have been sent. Instead we need to
+  // ensure that the error event is issued
+
+  expect(responseErrorListener).toHaveBeenCalledOnce()
+  expect(responseErrorListener).toHaveBeenCalledWith(
+    expect.objectContaining({ code: 'ERR_INVALID_ARG_TYPE' })
+  )
 
-  // Response stream errors are not request errors.
-  expect(requestErrorListener).not.toHaveBeenCalled()
-  expect(request.destroyed).toBe(false)
+  expect(requestErrorListener).toHaveBeenCalledOnce()
+  expect(requestErrorListener).toHaveBeenCalledWith(
+    expect.objectContaining({ code: 'ERR_INVALID_ARG_TYPE' })
```

Comment:
I don't believe this is correct. From what I've seen, once `response` is emitted, request can only be aborted if the response stream errors. This test implies that the request stream receives the same error as the response stream, but those are different streams and they do not share errors. Once the response headers are received, the request has done its work (well, technically, when the request stream is closed but you get the idea).